src/components/Rainbowcrack/Rainbowcrack.tsx

```diff
@@ -1,47 +1,82 @@
+import { SaveOutputToTextFile } from "../SaveOutputToFile/SaveOutputToTextFile";
 import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
 
-const title = "Rainbowcrack";
-const description_userguide =
-    "RainbowCrack is a computer program which utilises rainbow tables to be used in password cracking.\n" +
-    "How to use Rainbowcrack \n" +
-    "Step 1:Enter a hash value. (E.g. 5d41402abc4b2a76b9719d911017c592) \n " +
-    "Step 2:Simply tap on the button crack to crack the hash key. \n" +
-    "The user can even save the output to a file by assigning a file-name under 'save output to file' option.";
-
-interface FormValues {
-    hashcode: string;
+/**
+ * Represents the form values for the RainbowCrack component.
+ */
+interface FormValuesType {
+    hashValue: string;
 }
 
-export function rcrack() {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
+/**
+ * The RainbowCrack component.
+ * @returns The RainbowCrack component.
+ */
+const RainbowCrack = () => {
+    // Component State Variables
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+
+    // Component Constants
+    const title = "Rainbowcrack"; // Title of the component.
+    const description =
+        "RainbowCrack is a computer program which utilises rainbow tables to be used in password cracking."; // Description of the component.
+    const steps =
+        "How to use Rainbowcrack \n" +
+        "Step 1:Enter a hash value. (E.g. 5d41402abc4b2a76b9719d911017c592) \n" +
+        "Step 2:Simply tap on the button crack to crack the hash key. \n" +
+        "The user can even save the output to a file by assigning a file-name under 'save output to file' option."; // Steps to use the component.
+    const sourceLink = ""; // Link to the source code (or RainbowCrack documentation).
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["rainbowcrack"]; // Dependencies required by the component.
```

Comment:
This needs to be rcrack to match with the actual command. Otherwise the installationModal keeps popping up.